src/System.Threading/src/System/Threading/ReaderWriterLock.cs

```diff
@@ -0,0 +1,1339 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.ConstrainedExecution;
+
+namespace System.Threading
+{
+    public sealed class ReaderWriterLock : CriticalFinalizerObject
+    {
+        private const int InvalidThreadID = -1;
+        private const ushort MaxAcquireCount = ushort.MaxValue;
+        private static readonly int DefaultSpinCount = Environment.ProcessorCount != 1 ? 500 : 0;
+
+        private static long s_mostRecentLockID;
+
+        private ManualResetEvent _readerEvent;
+        private AutoResetEvent _writerEvent;
+        private long _lockID;
+        private volatile int _state;
+        private int _writerID = InvalidThreadID;
+        private int _writerSeqNum;
+        private ushort _writerLevel;
+
+#if RWLOCK_STATISTICS
+        private int _readerEntryCount;
+        private int _readerContentionCount;
+        private int _writerEntryCount;
+        private int _writerContentionCount;
+        private int _eventsReleasedCount;
+#endif // RWLOCK_STATISTICS
+
+        public ReaderWriterLock()
+        {
+            _lockID = Interlocked.Increment(ref s_mostRecentLockID);
+            GC.SuppressFinalize(this);
+        }
+
+        ~ReaderWriterLock()
```

Comment:
On second thought, I see a suppression was added in LocalDataStoreSlot:
```
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA1821", Justification = "Finalizer preserved for compat, it is suppressed by the constructor.")]
```
So I could do that here too, but it would be great if the finalizer stuff can be removed.